crates/op-rbuilder/src/integration/integration_test.rs

```diff
@@ -181,8 +181,34 @@ mod tests {
                 .send_raw_transaction(signed_tx.encoded_2718().as_slice())
                 .await?;
 
+            // Create a second reverting transaction
+            let tx_request = OpTypedTransaction::Eip1559(TxEip1559 {
+                chain_id: 901,
+                nonce: nonce + 2,
+                gas_limit: 300000,
+                max_fee_per_gas: base_fee.into(),
+                input: hex!("60006000fd").into(), // PUSH1 0x00 PUSH1 0x00 REVERT
+                ..Default::default()
+            });
+            let signed_tx = known_wallet.sign_tx(tx_request)?;
+            let reverting_tx_2 = provider
+                .send_raw_transaction(signed_tx.encoded_2718().as_slice())
+                .await?;
+
+            // Before block we should have 3 txs in pool, because they all valid
+            let pool = provider.txpool_status().await?;
+            assert_eq!(pool.pending, 3, "all txs should be in pending pool");
+            assert_eq!(pool.queued, 0, "queued pool should be empty");
+
             let block_hash = generator.generate_block().await?;
 
+            // TODO: uncomment once reth issue is addressed https://github.com/paradigmxyz/reth/issues/15973
```

Comment:
does this affect pending / queued pool metrics in reth as well?

Reply:
metrics are fine (checked it) 
Seems like only issue is on rpc call side

Reply:
okay, turned out that issue was with the test itself, found that problem and implementing the fix